### PR TITLE
Show message for style suggestions in the spellchecker

### PR DESCRIPTION
**Pull request by pablopunk**
SDBELGA-113

french spellchecker

### CODE REVIEW

File: scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx [hunk 1]
```diff
@@ -17,46 +21,71 @@ export class SpellcheckerContextMenuComponent extends React.Component<IProps> {
...
-                    <div className="form-label" style={{margin: '0 16px'}}>{gettext('Suggestions')}</div>
-                    {
-                        suggestions.length === 0
-                            ? <li><button>{gettext('SORRY, NO SUGGESTIONS.')}</button></li>
-                            : suggestions.map((suggestion, index) =>
+                <ul className={'dropdown__menu'} style={{ position: 'static' }}>
+                    <div className='form-label' style={{ margin: '0 16px' }}>
+                        {gettext('Suggestions')}
+                    </div>
+                    {messageExists && (
```
Comment (tomaskikutis) [inline scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx]: I see you've learned some angular :smile: 

You should never write multiple independent `if` statements for what is actually an `if/else`. When `if/else` is used, it's immediately clear that only one condition can run. When you use independent `if` statements, you remove that link and developer needs to analyze the conditions to understand that you really meant to write an `if/else`.

Use ternaries instead:

```
{
    messageExists
        ? (
            <div>a</div>
        )
        : (
            suggestions.length === 0
                ? (<div>a</div>)
                : (<div>b</div>)
        )
}
```

Nested ternaries might not look the nicest at the first glance either, but at least they represent the logic correctly.

File: scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx [hunk 2]
```diff
@@ -17,46 +21,71 @@ export class SpellcheckerContextMenuComponent extends React.Component<IProps> {
...
+                        <li>
+                            <button onMouseDown={() => this.onSuggestionClick(suggestions[0])}>{message}</button>
+                        </li>
+                    )}
+                    {!messageExists &&
```
Comment (tomaskikutis) [inline scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx]: Are you using boolean logic to simulate ternaries? :D why?

File: scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx [hunk 3]
```diff
@@ -91,4 +120,6 @@ export class SpellcheckerContextMenuComponent extends React.Component<IProps> {
     }
 }
 
-export const SpellcheckerContextMenu = connect()(SpellcheckerContextMenuComponent);
+export const SpellcheckerContextMenu = connect()(
```
Comment (tomaskikutis) [inline scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx]: Why change the formatting?

Comment (pablopunk) [reply scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx]: That was automatic, I'll revert it back

File: scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx (hunk 1 above)
Comment (pablopunk) [reply scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx]: At first it was not a ternary operator cause I didn't have the `!messageExist` part (it was showing the suggestions even if the message existed) so this was necessary. But yeah when I added the `!messageExists` part I didn't see the big picture

File: scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx (hunk 2 above)
Comment (pablopunk) [reply scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx]: See comment above

**Comment by pablopunk** [comment]
@tomaskikutis I made it so the message is shown in the loop if it exists, cause the behaviour is the same, check the diff again, it's way less code

**Pull request merged**